portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -53,6 +60,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get response from the form
     String text = request.getParameter("text-input");
+    String maxNumCommentsParam = request.getParameter("max-num");
+    if (maxNumCommentsParam != null && !maxNumCommentsParam.isEmpty()) {
+        maxNumComments = Integer.parseInt(request.getParameter("max-num"));
+    }
```

Comment:
For safety add an else block to fallback to some default number if the request was malformed. Alternatively return some error back to user.

Reply:
noted! thank you!

Reply:
noted! updated!